subsys/net/l2/openthread/openthread_utils.c

```
@@ -233,7 +233,7 @@ void add_ipv6_maddr_to_ot(struct openthread_context *context,
 	memcpy(&addr, addr6, sizeof(addr));
 
 	openthread_api_mutex_lock(context);
-	otIp6SubscribeMulticastAddress(context->instance, &addr);
+	(void)otIp6SubscribeMulticastAddress(context->instance, &addr);
```

This call is silently ignored:
```
/**
* Subscribes the Thread interface to a Network Interface Multicast Address.
*
* The passed in instance @p aAddress will be copied by the Thread interface. The Thread interface only
* supports a fixed number of externally added multicast addresses. See `OPENTHREAD_CONFIG_IP6_MAX_EXT_MCAST_ADDRS`.
*
* @param[in]  aInstance A pointer to an OpenThread instance.
* @param[in]  aAddress  A pointer to an IP Address.
*
* @retval OT_ERROR_NONE           Successfully subscribed to the Network Interface Multicast Address.
* @retval OT_ERROR_ALREADY        The multicast address is already subscribed.
* @retval OT_ERROR_INVALID_ARGS   The IP Address indicated by @p aAddress is an invalid multicast address.
* @retval OT_ERROR_REJECTED       The IP Address indicated by @p aAddress is an internal multicast address.
* @retval OT_ERROR_NO_BUFS        The Network Interface is already storing the maximum allowed external multicast
*                                 addresses.
*/
isn't it quite important to know that there are no more buffers left ?

Or at least comment why the return value, such as `OT_ERROR_NO_BUFS`, is ignored ?